src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,128 @@
+        match.addColorMatch(Color.kBlue); // blue
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    public DriverStation.Alliance GetcolorIntake() {
+        Color color = colorSensorIntake.getColor();
+        System.out.println(color.red + " Red " + color.blue + " Blue " + color.green + " Green ");
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+
+        if (resultColor == Color.kRed) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Color.kBlue) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    public DriverStation.Alliance GetcolorShooter() {
+        Color color = colorSensorShooter.getColor();
+        System.out.println(color.red + " Red " + color.blue + " Blue " + color.green + " Green ");
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+
+        if (resultColor == Color.kRed) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Color.kBlue) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    public int getCargoCount() {
+        return cargoCount;
+    }
+
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+//    public boolean Iscargoalliencecolor() {
+//        return color == Getcolor();
+//    }
+
+    public void setColor(DriverStation.Alliance fmsColor) {
+        color = fmsColor;
+    }
+
+    public DriverStation.Alliance getColorPosition(boolean first) {
+        if (first) {
+            return DriverStation.Alliance.valueOf(position.getFirst());
+        }
+        return DriverStation.Alliance.valueOf(position.getLast());
+    }
+
+    @Override
+    public void periodic() {
+        var colorShooter = GetcolorShooter();
+        var colorIntake = GetcolorIntake();
+        if (colorShooter != DriverStation.Alliance.Invalid) {
+            if (motor.getMotorOutputPercent() > 0) {
+                cargoCount--;
+                position.removeFirst();
+            } else if (motor.getMotorOutputPercent() < 0) {
+                cargoCount++;
+                position.addFirst(enumToString(colorShooter));
+            }
+
+        }
+        if (colorIntake != DriverStation.Alliance.Invalid) {
+            if (motor.getMotorOutputPercent() < 0) {
+                cargoCount--;
+                position.removeLast();
+            } else if (motor.getMotorOutputPercent() > 0) {
+                cargoCount++;
+                position.add(enumToString(colorIntake));

```

Comment:
Use `addLast()` instead to be consistent with `addFirst` (and also so it'd be clearer).